common/src/uint/endian.rs

```diff
@@ -40,42 +40,49 @@ macro_rules! define_slice_to_le {
     };
 }
 macro_rules! define_be_to_array {
-    ($name: ident, $type: ty, $byte_len: expr) => {
+    (@PRV $name: ident, $type: ty, $byte_len: expr) => {
         #[inline]
         pub fn $name(val: $type) -> [u8; $byte_len] {
-            debug_assert_eq!(::core::mem::size_of::<$type>(), $byte_len); // size_of isn't a constfn in 1.22
```

Comment:
Why remove the asserts?

Reply:
> Why remove the asserts?

'$byte_len' will now always be equal to `::core::mem::size_of::<$type>()` if the "public" variant of the macro is called (i.e. the non-`@PRV` one). To me it looked like the assert is somewhat redundant now, so I removed it. Do we want to protect ourselves against direct calls to the `@PRV` version of the macro?

Reply:
I don't see the harm... especially that any optimization will remove them.